File: ruby/etl/etl.rb
class ETL



  def self.transform(old)
    {}.tap do |output|
      old.each do |score, letters|
        output.merge!(self.transform_row(score,letters))
      end
    end
  end

  def self.transform_row(score, letters)
    (self.keys(letters).zip(self.values(score,letters))).to_h
  end

  def self.keys(letters)
    letters.map(&:downcase)
  end

  def self.values(score, letters)
    (1..letters.length).to_a.map{score}
  end

end

File: javascript/etl/etl.js
var ETL = function() {
  this.transform = function(old) {
    var newStyle = {};
    var values = Object.keys(old);
    for (var i = 0; i < values.length; i++) {
      var value = values[i];
      var keys = old[value];
      for(var j = 0; j < keys.length; j++) {
        var key = keys[j].toLowerCase();
        newStyle[key] = parseInt(value);
      }
    }
    return newStyle;
  }
};

module.exports = ETL;
